debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/DomainTypesIT.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.connector.postgresql;
+
+import java.sql.SQLException;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import io.debezium.connector.postgresql.PostgresConnectorConfig.SnapshotMode;
+import io.debezium.doc.FixFor;
+import io.debezium.junit.SkipWhenDatabaseVersion;
+
+import static io.debezium.junit.EqualityCheck.LESS_THAN;
+
+/**
+ * Integration test to verify behaviour of tables which include domain types
+ *
+ * @author Richard Kolkovich <richard@sigil.org>
+ */
+@SkipWhenDatabaseVersion(check = LESS_THAN, major = 11, minor = 0, reason = "Database Version less than 11")
+public class DomainTypesIT extends AbstractRecordsProducerTest {
+
+    @Before
+    public void before() throws SQLException {
+        TestHelper.dropAllSchemas();
+        TestHelper.execute("CREATE SCHEMA domaintypes");
+        TestHelper.execute("CREATE DOMAIN nmtoken AS text CHECK (VALUE ~* '^[A-Z0-9\\.\\_\\-\\:]+$');");
+        TestHelper.execute("CREATE TABLE domaintypes.t1 (id serial, token nmtoken, tokens nmtoken[]);");
+    }
+
+    @Test
+    @FixFor("DBZ-3657")
+    public void shouldInstallConnector() throws Exception {
+        start(PostgresConnector.class, TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.SNAPSHOT_MODE, SnapshotMode.INITIAL_ONLY)
+                .with(PostgresConnectorConfig.SCHEMA_INCLUDE_LIST, "domaintypes")
+                .build());
+        assertConnectorIsRunning();
```

Comment:
Can you add an assertion on an actual change event with the column type in question?

Reply:
Added, though note that the array type is not handled and therefore not in the record. The JDBC type for this is `2001`, which is unhandled in the base value converter.

Reply:
Ah yes, I see. In fact it's what I was wondering about when asking for a test. Thanks for clarifying. Any idea what it'd take to actually support arrays of domain types? The change in its current form is an improvement without doubt as those values will be gracefully ignored, but ideally of course we actually would capture and emit such value.

Reply:
So I've pushed a commit which at least allows to export such array columns of a domain type via `include.unknown.datatypes`, in its raw representation. It's not quite clear to me where this type 2001 is coming from (instead of whatever `TEXT` would be); otherwise the fallback to the proper schema and type would work. Anyways, better now than before.